### DISCUSSION TITLE
Test failure issue

### DISCUSSION THREAD

**Discussion by shankarpriyank** [oppia/oppia-android#4211] [Q&A (Installation)] [answered]
Ref[https://github.com/oppia/oppia-android/pull/4156](url) #4156 
I have trouble running tests locally 
I tried these solutions too but no help [https://stackoverflow.com/questions/51262364/kotlin-android-after-updating-kotlin-to-1-2-51-build-fails-execution-failed-f](url)

![image](https://user-images.githubusercontent.com/75121767/154965940-cf21b617-0d45-41f8-be75-cfe6439cebb0.png)

[https://gist.github.com/shankarpriyank/f1c76f07a4c866f838676fa889ec52e1](url)

**Comment by MaskedCarrot** [comment]
You can try `File` -> `Invalidate and Restart`. This might fix the issue.

**Reply by shankarpriyank** [reply]
Thanks 
I tried doing this but I still encounter the same error

**Reply by BenHenning** [reply]
@shankarpriyank can you include a Gist with the full error stack trace and link it in the main question? It's hard to determine context from a picture alone.

**Reply by shankarpriyank** [reply]
Done , please have a look , and tell me if I need to do anything else?

**Reply by BenHenning** [reply]
I see Caused by: java.lang.OutOfMemoryError: Java heap space in the output logs. I think you're running out of memory. How much memory do you have available?

**Reply by shankarpriyank** [reply]
I have 16gb Ram in my laptop , and I tweaked the settings a bit I am past this error now , but even now the tests donot run as intended , it shows a screen like this
![image](https://user-images.githubusercontent.com/75121767/155480471-01b3640e-e325-4fe4-9a1c-bf189abd4b8a.png)

**Reply by BenHenning** [reply]
@shankarpriyank have you tried running a whole test suite rather than a specific test method?

**Reply by shankarpriyank** [reply]
Ya, I have tried running a whole suite of tests like this
![image](https://user-images.githubusercontent.com/75121767/155689887-b2ea764c-f329-4e31-8c96-9327bd2b8e2e.png)
, and I get the same message "Test Passed 0 passed"
I even tried running the tests on a physical device but I got the same results

**Reply by BenHenning** [reply]
@shankarpriyank this might be an issue with your environment since I think you're trying to run instrumented tests. Can you try following the instructions here to see if Robolectric tests successfully run for you?

https://github.com/oppia/oppia-android/wiki/Oppia-Android-Testing#robolectric

**Reply by shankarpriyank** [reply]
I tried but was unable to  find that Android Junit Test in my Studio  it shows something like this 
![image](https://user-images.githubusercontent.com/75121767/155839749-40175377-39eb-47a4-93f6-6cdb4832d471.png)  
![image](https://user-images.githubusercontent.com/75121767/155839780-4f866139-ba1c-41c8-838a-c8b9610426b9.png)

**Reply by BenHenning** [reply]
Hmm I'm not sure. I might not be able to help further than this since I don't use Gradle in Android Studio. @ayush0402 or @anandwana001 do either of you have an idea on what might be going on here?

**Reply by anandwana001** [reply]
<img width="901" alt="Screenshot 2022-03-02 at 13 47 51" src="https://user-images.githubusercontent.com/25057618/156322468-d76b5056-95e8-447a-acb5-993bc4084ea5.png">


Are you following the correct way of running the test? @shankarpriyank

**Reply by shankarpriyank** [reply]
I was making a mistake while specifying the class , the test is running right now , will update what happens!

**Reply by shankarpriyank** [reply]
Again getting a bunch of errors

Firstly I got this error that the command line is too long 
![Command line](https://user-images.githubusercontent.com/75121767/156348295-aca2bad4-52e7-40aa-aecd-6848f94b381b.png)
 But then I went through this and tried enabling the option of shortening the command line [https://stackoverflow.com/questions/47926382/how-to-configure-shorten-command-line-method-for-whole-project-in-intellij](url)
And I tweaked my configuration a bit 
![image](https://user-images.githubusercontent.com/75121767/156348706-bfed419d-b0ac-4f11-af42-e4e63864e42f.png)


 And I then ran into this error [https://gist.github.com/shankarpriyank/8bd90c7baa321f53e21bcae652c4540d](url)
![image](https://user-images.githubusercontent.com/75121767/156348045-a0782e91-8599-423a-b85c-16c94759c69e.png)

**Reply by anandwana001** [reply]
Are you using the correct Java 8? Use the one which has an Android platform in it. 
It will give options to choose from different Android platforms.
@shankarpriyank

**Reply by shankarpriyank** [reply]
Thanks, @anandwana001 the tests are running now
And @BenHenning actually all of them passed!
![image](https://user-images.githubusercontent.com/75121767/156387105-b09e6654-ec34-4c1a-942a-92d232ca1db3.png)
![image](https://user-images.githubusercontent.com/75121767/156389319-13ee965c-4c4f-40bc-82be-f6d32f0d9979.png)

**Comment by anandwana001** [comment, marked as answer]
Answer - https://github.com/oppia/oppia-android/discussions/4211#discussioncomment-2278831
